docs/source/reference/pysupported.rst

```diff
@@ -117,6 +128,65 @@ Coroutine features of generators are not supported (i.e. the
 :meth:`generator.send`, :meth:`generator.throw`, :meth:`generator.close`
 methods).
 
+.. _pysupported-exception-handling:
+
+Exception handling
+------------------
+
+``raise`` statement
+'''''''''''''''''''
+
+The ``raise`` statement is only supported in the following forms:
+
+* ``raise SomeException``
+* ``raise SomeException(<arguments>)``: in :term:`nopython mode`, constructor
+  arguments must be :term:`compile-time constants <compile-time constant>`
+
+It is currently unsupported to re-raise an exception created in compiled code.
+
+``try .. except``
+'''''''''''''''''
+
+The ``try .. except`` construct is partially supported. The following forms
+of are supported:
+
+* the *bare* except that captures all exceptions:
+
+  .. code-block:: python
+
+    try:
+        ...
+    except:
+        ...
+
+* matching against exactly the *Exception* class:
```

Comment:
When I read this, I took it to mean that subclasses of `Exception` are not matched. However, when I tried:

```python
from numba import njit

def exc_test():
    try:
        raise RuntimeError
    except Exception:
        return 1
    else:
        return 2

njit_exc_test = njit(exc_test)

print("CPython exc test  =", exc_test())
print("Nopython exc test =", njit_exc_test())
```

I got:

```
CPython exc test  = 1
Nopython exc test = 1
```

whereas I was expecting based on the wording:

```
CPython exc test  = 1
Nopython exc test = 2
```

I also tried:

```python
from numba import njit

class ExampleError(Exception):
    pass

def exc_test():
    try:
        raise ExampleError
    except ExampleError:
        return 1
    else:
        return 2

njit_exc_test = njit(exc_test)
```

Which I was thinking might work, based on the fact that catching `RuntimeError` above worked - however the compile failed (as I imagine you would expect).

Would it be clearer to say something along the lines of "matching against the `Exception` class - matching against other classes is not supported. Note that when instances of `Exception` subclasses are raised, they will also be matched by `except Exception`, as they would in non-jitted code"?

Reply:
Thanks for reviewing it. You're right that the sentence is unclear. I'll update it.